llvm-hs/src/Control/Monad/Trans/AnyCont.hs

```diff
@@ -17,22 +18,26 @@ instance Applicative (AnyContT m) where
   pure a = AnyContT $ pure a
   f <*> v = AnyContT $ unAnyContT f <*> unAnyContT v
 
-instance Monad m => Monad (AnyContT m) where
+instance MonadFail m => Monad (AnyContT m) where
```

Comment:
I’m not really happy about this. It seems like we add a `MonadFail` constraint now only to remove it once `fail` is moved out of Monad. Is there a downside to keeping the existing implementation of `fail` in the `Monad` instance? That would avoid having to introduce a `MonadFail m` constraint wherever `AnyContT` is used and seems to be consistent with how the `transformers` library handles this.

Reply:
I agree, it is not a very nice change (but, it is the recommended one, so I just did it...)

How about this implementation instead? https://github.com/tmcdonell/llvm-hs/commit/1aaa85ade0b5723d79ad7372642d3d5ffa07e942

Reply:
That’s what I had in mind and it looks way better! I don’t think the recommendation really applies to transformers.

Reply:
Cool. Updated this PR to use that patch instead.